examples/usvg_viewer/src/asset.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::{bail, Result};
+use generic_array::GenericArray;
+use hex_literal::hex;
+use sha2::{Digest, Sha256};
+use std::env::temp_dir;
+use std::io::Read;
+use std::path::PathBuf;
+
+pub struct SvgAsset {
+    url: &'static str,
+    sha256sum: [u8; 32],
+    pub license: &'static str,
+    pub size: u128,
+}
+
+impl SvgAsset {
+    pub fn local_path(&self) -> PathBuf {
+        let arr = GenericArray::from(self.sha256sum);
+        temp_dir()
+            .join(format!("vello-asset-{:x}", arr))
+            .with_extension("svg")
+    }
+
+    pub fn fetched(&self) -> bool {
+        let resource_local_path = self.local_path();
+        if let Ok(contents) = std::fs::read_to_string(&resource_local_path) {
+            if Sha256::digest(contents)[..] == self.sha256sum {
```

Comment:
The assets refer to specific SVG images. You can easily edit the downloaded files and load them with the viewer. They just won't be the builtin assets anymore. They'll be different images.

Reply:
Mostly, I also don't like us using the temporary directory for them.

My preference would be to put them in e.g. `examples/assets/downloads`. They're not *huge* files, so them not being not cleaned up automatically is not a huge issue.

Reply:
What should happen if `usvg_viewer` is called from outside the repository root?

Reply:
It should use the same directory?

These aren't meant to be portable binaries. So we can just find the directory, relative to the (compile time) CARGO_MANIFEST_DIR

Reply:
How about we add the public domain images to the assets in the repository? Their combined size is 7.3MiB (optimized and compressed).

Or maybe it would be better for the SVG viewer to live in a separate repository.

Reply:
The issue with having them in a subdirectory is that it grows the entire history; so not only do you need to download them once - even if/when we remove them or rotate them, all clones still need to download them.

I think having the assets in a submodule could reaosnably make sense, although that has the issue of requiring submodule management, which is a massive pain.

Overall, I think downloading into a non-tracked subdirectory is the least bad option, although I'm not especially thrilled by it.

My thoughts would be that a dedicated svg viewer program more complicated than "render SVG undecorated" doesn't really belong in this repository (e.g. it should probably be using Xilem instead, once that's a bit more ready)